aws-crt-kotlin/native/src/aws/sdk/kotlin/crt/Allocator.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.sdk.kotlin.crt
+
+import kotlinx.cinterop.*
+import libcrt.*
+
+internal object Allocator {
+    val Default: AwsAllocator = AwsAllocator()
+}
+
+@OptIn(ExperimentalForeignApi::class)
+internal class AwsAllocator : NativeFreeablePlacement, CValuesRef<aws_allocator>() {
```

Comment:
I wonder if this can just be `internal object AwsAllocator` and then we don't need to reference `Allocator.Default`. Leave it for now I can mess with it when I toy with memory management stuff